src/spanningtrees/boruvka.jl

```diff
@@ -0,0 +1,73 @@
+"""
+    boruvka_mst(g, distmx = weights(g); minimize = true)
+
+Return a tuple `(mst, weights)` where `mst` is a vector of edges representing the
+optimum (minimum, by default) spanning tree of a connected, undirected graph
+`g` with optional matrix `distmx` that provides distinct edge weights, and
+`weights` is the sum of all the edges in the solution by using
+[Boruvka's algorithm](https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm).
+The algorithm requires that all edges have different weights to correctly generate a minimun/maximum spanning tree
+### Optional Arguments
+- `minimize=true`: if set to `false`, calculate the maximum spanning tree.
+"""
+
+function boruvka_mst end 
+
+@traitfn function boruvka_mst(graph::AG::(!IsDirected), 
+        distmx::AbstractMatrix{T} = weights(graph); 
+        minimize = true) where {T<:Real,U,AG<:AbstractGraph{U}}
+
+    djset = IntDisjointSets(nv(graph))
+    # maximizing Z is the same as minimizing -Z
+    # mode will indicate the need for the -1 multiplication
+    mode = (-1)^(minimize ? 0 : 1)
```

Comment:
This is a bit awkward. Why not:
```
mode = minimize ? 1 : -1
```

Reply:
I need the variable mode to be 1 or -1, if I use 1 : -1 we have (-1)^1 = -1 and (-1)^(-1) that is also equals -1, so it doesn't work.

Reply:
Done, thanks!